src/Nethermind/Nethermind.Evm/CodeAnalysis/JumpDestinationAnalyzer.cs

```diff
@@ -98,25 +103,167 @@ private void AnalyzeJumpDestinations(out object previous)
+            CreateJumpDestinationBitmap_Vector512(bitmap, code) :
+            Vector128<sbyte>.IsSupported && code.Length >= Vector128<sbyte>.Count ?
+            CreateJumpDestinationBitmap_Vector128(bitmap, code) :
+            CreateJumpDestinationBitmap_Scalar(bitmap, code);
+    }
+
+    internal static long[] CreateBitmap(int codeLength)
+        => new long[GetInt64ArrayLengthFromBitLength(codeLength)];
+
+    [SkipLocalsInit]
+    internal static long[] CreateJumpDestinationBitmap_Scalar(long[] bitmap, ReadOnlySpan<byte> code)
+    {
+        JumpDestinationBitmap_Scalar(programCounter: 0, bitmap, code);
+        return bitmap;
+    }
+
+    [SkipLocalsInit]
+    private static void JumpDestinationBitmap_Scalar(nuint programCounter, Span<long> bitmap, ReadOnlySpan<byte> code)
+    {
+        // We accumulate each array segment to a register and then flush to memory when we move to next.
+        long currentFlags = 0;
+        while (programCounter < (nuint)code.Length)
+        {
+            // Grab the instruction from the code; zero length code
+            // doesn't enter this method and we check at end of loop if
+            // hit the last element and should exit, so skip bounds check
+            // access here.
+            int op = Unsafe.AddByteOffset(ref MemoryMarshal.GetReference(code), programCounter);
+            // Set default programCounter increment to 1 for default case when we don't read a PUSH.
+            nuint move = 1;
+
+            if (op == JUMPDEST)
+            {
+                // Accumulate Jump Destinations to register, shift will wrap and single bit
+                // so can shift by the whole programCounter.
+                currentFlags |= 1L << (int)programCounter;
+            }
+            else if ((sbyte)op > PUSHx)
+            {
+                // Fast forward programCounter by the amount of data the push
+                // represents as don't need to analyze data for Jump Destinations.
+                move = (nuint)op - PUSH1 + 2;
+            }
+
+            nuint nextCounter = programCounter + move;
+            // Does the move mean we are moving to new segment of the long array?
+            // If we take the current index in flags, and add the move, are we at
+            // a new long segment, i.e. a larger than 64 position move.
+            if (currentFlags != 0 && ((programCounter & 63) + move >= 64 || nextCounter >= (nuint)code.Length))
+            {
+                // Moving to next array element (or finishing) assign to array.
+                MarkJumpDestinations(bitmap, programCounter, currentFlags);
+                // Clear the flags in preparation for the next array segment.
+                currentFlags = 0;
+            }
+
+            // Move to next instruction.
+            programCounter = nextCounter;
+        }
+    }
+
+    [SkipLocalsInit]
+    internal static long[] CreateJumpDestinationBitmap_Vector512(long[] bitmap, ReadOnlySpan<byte> code)
```

Comment:
it has `Create` but doesn't create anything?

Reply:
Rnamed